resolver_service/app.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from flask import Flask
+import logging.config
+from flask_restful import Api
+from flask_discoverer import Discoverer
+from .models import Record
+from .views import bp
+from sqlalchemy.orm import load_only
+
+def create_app(config=None):
+    """
+    Create the application and return it to the user
+    :return: flask.Flask application
+    """
+
+    app = FlaskApp(__name__, static_folder=None)
+    app.url_map.strict_slashes = False
+
+    # Load config and logging
+    load_config(app, config)
+    logging.config.dictConfig(
+        app.config['RESOLVER_SERVICE_LOGGING']
+    )
+
+    # Register extensions
+    api = Api(app)
+    Discoverer(app)
+
+    app.register_blueprint(bp)
+    return app
+
+
+def FlaskApp(Flask):
+
+    def __init__(self, *args, **kwargs):
+        super(Flask, self).__init__(*args, **kwargs)
+
+    
+    def get_record(self, bibcode, link_type, link_subtype='', db='nonbib', fields=None):
```

Comment:
notice one method replaces the initial two methods; i've also removed all the config (also for messages); they were saying the same thing - so the advice in general is: do not repeat yourself

also, and this is very important - you were initially creating engine/connection for every call ofthe function - this should be avoided, it is very expensive

so now it is inside an applicaiton object

also, I've removed the pure sql queries and replaced them with sqlalchemy orm (that way we can use any database engine we like and is supported by sqa)

Reply:
I haven't tested any of the code, but I'm confident it should work with those additions:

- you will need to review and update models.py
- create alembic recipe (every microservice needs to have it), see: https://github.com/adsabs/solr-service/tree/master/alembic

Reply:
need to understand your changes, and so I am starting with my own version and adding your suggestions that makes sense to me and I understand. the ones I don't understand then we can have a discussion about them.